api/src/main/java/org/openmrs/module/ugandaemrreports/reports2019/SetupMER_TX_TBReport.java

```diff
@@ -0,0 +1,186 @@
+
+        CohortDefinition above15Years = cohortDefinitionLibrary.above15Years();
+        CohortDefinition below15Years = cohortDefinitionLibrary.agedAtMost(14);
+
+        CohortDefinition screenedNegative = hivCohortDefinitionLibrary.getScreenedForTBNegativeDuringPeriod();
+        CohortDefinition screenedPositive  = df.getPatientsNotIn(hivCohortDefinitionLibrary.getAssessedForTBDuringPeriod(),screenedNegative);
+
+        CohortDefinition newOnARTAndStartedTBTreatmentDuringPeriod = df.getPatientsInAll(havingArtStartDateDuringQuarter,startedTBTreatmentDuringPeriod);
+        CohortDefinition previouslyOnARTAndStartedTBTreatmentDuringPeriod = df.getPatientsInAll(havingArtStartDateBeforeQuarter,startedTBTreatmentDuringPeriod);
+
+        addGender(dsd,"1","new on ART start TB treatment below 15 years ", df.getPatientsInAll(newOnARTAndStartedTBTreatmentDuringPeriod,below15Years));
+        addGender(dsd,"2","new on ART start TB treatment above 15 years ", df.getPatientsInAll(newOnARTAndStartedTBTreatmentDuringPeriod,above15Years));
+
+        addGender(dsd,"3","previously on ART start TB treatment below 15 years ", df.getPatientsInAll(previouslyOnARTAndStartedTBTreatmentDuringPeriod,below15Years));
+        addGender(dsd,"4","previously on ART start TB treatment above 15 years ", df.getPatientsInAll(previouslyOnARTAndStartedTBTreatmentDuringPeriod,above15Years));
+
+
+        addGender(dsd,"5","new on ART start TB treatment screen positive below 15 years ", df.getPatientsInAll(havingArtStartDateDuringQuarter,screenedPositive,below15Years));
+        addGender(dsd,"6","new on ART start TB treatment screen positive above 15 years ", df.getPatientsInAll(havingArtStartDateDuringQuarter,screenedPositive,above15Years));
+
+        addGender(dsd,"7","new on ART start TB treatment screen negative below 15 years ", df.getPatientsInAll(havingArtStartDateDuringQuarter,screenedNegative,below15Years));
+        addGender(dsd,"8","new on ART start TB treatment screen negative above 15 years ", df.getPatientsInAll(havingArtStartDateDuringQuarter,screenedNegative,above15Years));
+
+        addGender(dsd,"9","previously on ART start TB treatment screen positive below 15 years ", df.getPatientsInAll(havingArtStartDateBeforeQuarter,screenedPositive,below15Years));
+        addGender(dsd,"10","previously on ART start TB treatment screen positive above 15 years ", df.getPatientsInAll(havingArtStartDateBeforeQuarter,screenedPositive,above15Years));
+
+        addGender(dsd,"11","previously on ART start TB treatment screen negative below 15 years ", df.getPatientsInAll(havingArtStartDateBeforeQuarter,screenedNegative,below15Years));
+        addGender(dsd,"12","previously on ART start TB treatment screen negative above 15 years ", df.getPatientsInAll(havingArtStartDateBeforeQuarter,screenedNegative,above15Years));
+
+
+
+        return rd;
+    }
+
+    public CohortDefinitionDimension getGender(){
```

Comment:
@dbaluku Why not put this in the CohortDefinitionDimension library to be used in other reports

Reply:
This is mapping report parameters, this cant be general